src/DynamoCoreWpf/ViewModels/Search/NodeSearchElementViewModel.cs

```diff
@@ -286,11 +286,12 @@ protected virtual void CreateAndConnectToPort(object parameter)
             dynamoViewModel.ExecuteCommand(new DynamoModel.CreateAndConnectNodeCommand(id, initialNode.GUID,
                 Model.CreationName, 0, portModel.Index, adjustedX, 0, createAsDownStreamNode, false, true));
 
-            // Clear current selections and select all input nodes as we need to perform Auto layout on only the input nodes.
-            DynamoSelection.Instance.ClearSelection();
             var inputNodes = initialNode.InputNodes.Values.Where(x => x != null).Select(y => y.Item2);
 
+            // Clear current selections and select all input nodes as we need to perform Auto layout on only the input nodes.
+            DynamoSelection.Instance.ClearSelection();
             DynamoSelection.Instance.Selection.AddRange(inputNodes);
+            DynamoSelection.Instance.Selection.Add(initialNode);
```

Comment:
Doesn't this cause the original node to jump during auto-layout?

Reply:
No, because we are skipping the original node while saving coordinates.

Reply:
Then why do you need to include it here?

Reply:
You're skipping it anyway.